avocado/utils/process.py

```diff
@@ -808,14 +808,58 @@ def poll(self):
             self._fill_results(rc)
         return rc
 
-    def wait(self):
+    def wait(self, timeout=None, sig=signal.SIGTERM):
         """
         Call the subprocess poll() method, fill results if rc is not None.
+
+        :param timeout: Time (seconds) we'll wait until the process is
+                        finished. If it's not, we'll try to terminate it
+                        and it's children using ``sig`` and get a
+                        status. When the process refuses to die
+                        within 1s we use SIGKILL and report the status
+                        (be it exit_code or zombie)
+        :param sig: Signal to send to the process in case it did not end after
+                    the specified timeout.
         """
+        def nuke_myself():
+            self.result.interrupted = ("timeout after %ss"
+                                       % (time.time() - self.start_time))
+            try:
+                kill_process_tree(self.get_pid(), sig, timeout=1)
+            except Exception:
```

Comment:
The logging object is already available here in this module. I think it's a good idea to log both those exceptions case at least with as an warning.

Reply:
Sure, makes sense.